### DISCUSSION TITLE
Using model-extra-attributes 'serde_with::skip_serializing_none'

### DISCUSSION THREAD

**Discussion by cm-auto** [SeaQL/sea-orm#1655] [Q&A] [answered]
## Motivation

To use external attributes, we have to import their crates.

When files are generated using `sea-orm-cli generate entity --with-serde both`, then the import statement `use serde::{Deserialize, Serialize};` is included in the generated file.

However if I try to add an extra attribute to a model with for example `sea-orm-cli generate entity --model-extra-attributes 'skip_serializing_none'` it successfully adds the attribute, but the compilation of my project fails and it says the following:
```
cannot find attribute `skip_serializing_none` in this scope
consider importing this attribute macro:
serde_with::skip_serializing_none
```

I can of course manually add the import, but I would prefer it to happen automatically.

## Proposed Solutions

Adding the possibility to specify which crates to import when generating files with sea-orm-cli generate entity.
For this sea-orm-cli should take a new argument: "--extra-use-statements", which takes the import statements as value.
Example:
```bash
sea-orm-cli --extra-use-statements 'serde_with::skip_serializing_none'
```

## Additional Information

I actually already made a fork and implemented this behavior:
https://github.com/SeaQL/sea-orm/commit/27591239fd13d164efd689b6b36e649d035c0e76

I'm aware that I could just do this to solve my problem:
```bash
sea-orm-cli generate entity --model-extra-attributes 'serde_with::skip_serializing_none'
```
However when I realized I could do this, I already implemented this behavior and I thought this might be useful for something else.

I ran the tests of "sea-orm", "sea-orm-cli" and "sea-orm-codegen" before and after adding my changes. Everything succeeded, except for one test of "sea-orm-cli", but it also failed before adding my changes.

I forked from this commit https://github.com/SeaQL/sea-orm/commit/9f6b5664eba59c5eafeb6ce91e8946af47983fdd

**Comment by tyt2y3** [comment, marked as answer]
```sh
sea-orm-cli generate entity --model-extra-attributes 'serde_with::skip_serializing_none'
```

Thank you for your sharing
